src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -7322,6 +7279,1521 @@ public Test(int b1, int b2, int b3)
+        {
+            var template = """
+{0}
+
+class Test
+{{
+    public void Test2()
+    {{
+        var d = {1} (ref {4}, {2}) =>
+        {{
+            {3}
+            return default;
+        }};
+    }}
+}}
+""";
+
+            VerifyParameterRefSafetyScope(template, memberName: "Test.<>c.<Test2>b__0_0");
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_08_Lambda_Delegate()
+        {
+            var template = """
+{0}
+
+class Test
+{{
+    public void Test2()
+    {{
+        var d = {1} (ref {4}, {2}) =>
+        {{
+            {3}
+            return default;
+        }};
+    }}
+}}
+""";
+
+            VerifyParameterRefSafetyScope(template, memberName: "<>f__AnonymousDelegate0.Invoke");
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_09_LocalFunction()
+        {
+            var template = """
+{0}
+
+class Test
+{{
+    public void Test2()
+    {{
+        var d = local;
+
+        {1} local (ref {4}, {2})
+        {{
+            {3}
+            return default;
+        }};
+    }}
+}}
+""";
+
+            VerifyParameterRefSafetyScope(template, memberName: "Test.<Test2>g__local|0_0");
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_10_Mismatch_Overriding()
+        {
+            var src = @"
+using System;
+
+abstract class C1
+{
+    public abstract Span<long> Test1(Span<long> a);
+    public abstract Span<long> Test2(Span<long> a);
+    public abstract Span<long> Test3(Span<long> a);
+}
+
+class C2 : C1
+{
+    public override Span<long> Test1(params Span<long> a)
+        => throw null;
+
+    public override Span<long> Test2(params scoped Span<long> a)
+        => throw null;
+
+    public override Span<long> Test3(scoped Span<long> a)
+        => throw null;
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            Assert.Equal(comp.GetMember<MethodSymbol>("C1.Test1").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test1").Parameters.Single().EffectiveScope);
+            Assert.False(comp.GetMember<MethodSymbol>("C2.Test1").Parameters.Single().IsParams);
+            Assert.NotEqual(comp.GetMember<MethodSymbol>("C1.Test2").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test2").Parameters.Single().EffectiveScope);
+            Assert.NotEqual(comp.GetMember<MethodSymbol>("C1.Test3").Parameters.Single().EffectiveScope, comp.GetMember<MethodSymbol>("C2.Test3").Parameters.Single().EffectiveScope);
+
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_11_Mismatch_Overriding()
+        {
+            var src = @"
+using System;
+
+abstract class C1
+{
+    public abstract Span<long> Test1(params Span<long> a);
+    public abstract Span<long> Test2(params scoped Span<long> a);
+    public abstract Span<long> Test3(scoped Span<long> a);
+}
```

Comment:
The language doesn't report errors about a 'scoped' mismatch across base/override, interface/impl, delegate/method group, etc. relationships when the signature does not return any references. For example, for a method `abstract void M(scoped Span<int> s)`, we don't care whether the override has a `scoped` modifier, as the method signature doesn't have allow the possibility of returning references and therefore the lack of `scoped` won't violate ref safety. A method like `abstract Span<int> M(scoped Span<int> s)` on the other hand *will* care if the override is missing a `scoped` modifier.

So, it might make the tests more complete, in that they will verify that ref safety is checked across related signatures, if some methods in the tests are modified to also be able to return references, through ref struct or by-ref return type.

One example of a test which might be meaningful is:

```cs
using System;
using System.Diagnostics.CodeAnalysis;

class C1
{
    public virtual Span<long> Test1(scoped Span<long> a) => throw null!;
}

class C2 : C1
{
    // expect CS8987: The 'scoped' modifier of parameter 'a' doesn't match overridden or implemented member.
    public override Span<long> Test1(Span<long> a) => throw null!;
}

class C3 : C1
{
    // ok
    public override Span<long> Test1(scoped Span<long> a) => throw null!;
}

class C4 : C1
{
    // ok
    public override Span<long> Test1(params Span<long> a) => throw null!;
}

class C5 : C1
{
    // expect CS8987: The 'scoped' modifier of parameter 'a' doesn't match overridden or implemented member.
    public override Span<long> Test1([UnscopedRef] params Span<long> a) => throw null!;
}
```

It's OK with me if it's not a goal of this PR to test this aspect of ref safety on signatures. But, I wanted to make sure that if it *is* a goal, that this aspect is tested.

Reply:
> The language doesn't consider it an error to have a 'scoped' mismatch across base/override, interface/impl, delegate/method group, etc. relationships when the signature does not return any references. 

I do not think this aspect is affected by the feature. Therefore, I do not find it compelling to test that. Nothing changes in terms of determining whether signature returns any references, I think.

Reply:
It looks like I had a misunderstanding, which arose from my review tool seeing an outdated version of the PR. I'm satisfied with the newer version of the tests. Thanks